options/options.go

```diff
@@ -452,6 +461,47 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 	return args, err
 }
 
+// ParseConfigFile iterates over args to find a --config option. If not found, we return.
+// If found, we read the contents of the specified config file in YAML format. We parse
+// any values corresponding to --password, --uri and --sslPEMKeyPassword, and store them
+// in the opts.
+func (opts *ToolOptions) ParseConfigFile(args []string) error {
+	// Get config file path from the arguments, if specified.
+	_, err := opts.parser.ParseArgs(args)
+	if err != nil {
+		return err
+	}
+
+	// No --config option was specified.
+	if opts.General.ConfigPath == "" {
+		return nil
+	}
+
+	// --config option specifies a file path.
+	configBytes, err := ioutil.ReadFile(opts.General.ConfigPath)
+	if err != nil {
+		return errors.Wrapf(err, "error opening file with --config")
+	}
+
+	// Unmarshal the config file as a top-level YAML file.
+	var config struct {
+		Password          string `yaml:"password"`
+		ConnectionString  string `yaml:"uri"`
+		SSLPEMKeyPassword string `yaml:"sslPEMKeyPassword"`
+	}
+	err = yaml.UnmarshalStrict(configBytes, &config)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing config file %s", opts.General.ConfigPath)
+	}
+
+	// Assign each parsed value to its respective ToolOptions field.
+	opts.Auth.Password = config.Password
+	opts.URI.ConnectionString = config.ConnectionString
+	opts.SSL.SSLPEMKeyPassword = config.SSLPEMKeyPassword
```

Comment:
To bring up a comment I made in the tech design a while ago -- before storing each field here, we could actually check to see if the fields were already set by the call to `opts.parser.ParseArgs()` at the beginning of this func. This would allow us to log a message that a config file's value will be overridden by the respective command line option.

Might offer a bit of transparency but I don't feel strongly either way.

Reply:
I don't recommend doing that because the fields aren't necessarily directly set inside cmd parameters, for example, the password could be set inside connectionString. In that case, you won't be able to catch the overridden. 
If you really wanna log the warning message, this function should return a  separate ToolOptions object and later used as the comparison with the `ParseArgs()` result.

Reply:
Right, it's true we wouldn't know if a password comes from `--uri` vs. a positional connection string, but would that matter a lot?

It might be enough to just say something like `warning: connection string in config file will be overridden by connection string on command line`.

Reply:
Since we already decided this wasn't a requirement for this project, I would also recommend that we not do this